src/main/webapp/script.js

```diff
@@ -0,0 +1,162 @@
+      electionList.forEach((election) => {
+        let ocdId = election.ocdDivisionId;
+        let stateId = ocdId.indexOf(OCD_STATE_TITLE);
+        let districtId = ocdId.indexOf(OCD_DISTRICT_TITLE);
+
+        // ocdDivisionId is in the form "ocd-division/country:us/state:<state_code>"
+        // For example, Lousiana's ID is "ocd-division/country:us/state:la"
+        if (stateId !== -1) {
+          // If the state text is found in ocdDivisionId, then this is a state election.
+          if (ocdId.substring(stateId, stateId + OCD_STATE_TITLE.length + STATE_CODE_LENGTH)
+              == selectedStateId) {
+            stateElections.push(election);
+          }
+        } else if (districtId !== -1) {
+          // If the district text is found in ocdDivisionId, then this is an election in DC.
+          if (ocdId.substring(districtId, districtId + OCD_DISTRICT_TITLE.length + STATE_CODE_LENGTH) 
+              == selectedStateId) {
+            stateElections.push(election);
+          }
+        } else {
+          // No state or district text found, so this is a national election
+          nationalElections.push(election);
+        }
+      });
+      
+      const electionListContainerElement = document.getElementById('election-list-content');
+      electionListContainerElement.innerHTML = '';
+      
+      const electionsInStateHeadingElement = document.createElement('h3');
+      electionsInStateHeadingElement.innerHTML = ELECTION_LIST_TITLE + selectedStateName;
+      electionListContainerElement.appendChild(electionsInStateHeadingElement);
+
+      if (stateElections.length == 0) {
+        buildEmptyListText(electionListContainerElement);
+      } else {
+        buildElectionListElements(electionListContainerElement, stateElections);
+      }
+
+      const electionsInNationHeadingElement = document.createElement('h3');
+      electionsInNationHeadingElement.innerHTML = ELECTION_LIST_TITLE + COUNTRY_NAME;
+      electionListContainerElement.appendChild(electionsInNationHeadingElement);
+
+      if (nationalElections.length == 0) {
+        buildEmptyListText(electionListContainerElement);
+      } else {
+        buildElectionListElements(electionListContainerElement, nationalElections);
+      }
+  });
+}
+
+/**
+ * Build the HTML elements that make up the list of upcoming elections.
+ * @param {HTMLDivElement} containerElement div container for election list
+ * @param {JSON} elections JSON object containing the list of relevant elections.
+ */
+function buildElectionListElements(containerElement, elections) {
+  const electionListElement = document.createElement('ul');
+  electionListElement.className = 'list-group list-group-flush';
+  electionListElement.id = 'election-list';
+
+  elections.forEach((election) => {
+    const electionItemElement = document.createElement('li');
+    electionItemElement.className = 'list-group-item';
+    electionItemElement.id = 'election-item';
```

Comment:
I thought we're not allowed to have multiple elements with the same html id?

Reply:
Oops I should've made that a class
Not sure why it still works hahah but I'll change it!